### ISSUE TITLE
Support for ignoring lines

### ISSUE THREAD

**Issue by danallen88** [VeryGoodOpenSource/very_good_coverage#200]
Hello - wanted to ask the question here since I can't seem to find a related issue in Dart's coverage docs. Is there support for ignoring a single line or group of lines with this action?  With the following block, I am able to have coverage at 100% locally, but it is showing the lines within the if-statement are not covered when run with this action.

```dart
// coverage:ignore-start
if (variableName.length > 1) {
    _logger.debug(
        'My logging statement here with ${variableName.length} added '
        'in for additional information');
}
// coverage:ignore-end
```

Command run locally that works with 100% coverage on very_good version 0.9.1:
```sh
very_good test -j 28 -x presubmit-only --coverage --min-coverage 100 --test-randomize-ordering-seed random --no-optimization --exclude-coverage "**/{*.g.dart,l10n/*.dart,l10n/**/*.dart,main/bootstrap.dart,*.gen.dart}"
```

We're using VeryGoodOpenSource/very_good_coverage@v2 for the GitHub action.

**Comment by felangel** [comment]
Hey @danallen88 👋 
Thanks for opening an issue!

What version of Dart/Flutter are you using? Can you also confirm that you're using the same version of Dart locally and in CI?

**Comment by danallen88** [comment]
Hi @felangel! Locally I have Flutter 3.3.0 and Dart 2.18.0.  

On CI there is Dart 2.18.6, and this is a pure Dart package with no Flutter dependency.

**Comment by felangel** [comment]
> Hi @felangel! Locally I have Flutter 3.3.0 and Dart 2.18.0.
> 
> On CI there is Dart 2.18.6, and this is a pure Dart package with no Flutter dependency.

Can you try upgrading to Dart 2.18.6 locally and see if you're able to reproduce?

**Comment by danallen88** [comment]
> Can you try upgrading to Dart 2.18.6 locally and see if you're able to reproduce?

Yes I just ran the same command locally using Dart 2.18.6 and was not able to reproduce the same issue the CI action has.

**Comment by felangel** [comment]
Can you share your CI workflow? It might be a discrepancy between the version of package:coverage.

**Comment by felangel** [comment]
> ```yaml
> name: my_repository
> 
> concurrency:
>   group: ${{ github.workflow }}-${{ github.ref }}
>   cancel-in-progress: true
> 
> on:
>   pull_request:
>     paths:
>       - "packages/my_repository/**"
>       - ".github/workflows/my_repository.yaml"
>     branches:
>       - main
> 
> jobs:
>   build:
>     uses: VeryGoodOpenSource/very_good_workflows/.github/workflows/dart_package.yml@v1
>     with:
>       working_directory: packages/my_repository
> ```

Thanks! How are you verifying coverage in CI?

**Comment by danallen88** [comment]
> Thanks! How are you verifying coverage in CI?

We're using the code coverage component of the Dart package Very Good Workflow: [IdexxThumbnailStackThemeData](https://github.com/VeryGoodOpenSource/very_good_workflows#dart-package-workflow)

**Comment by danallen88** [comment]
I'm not sure if it helps or if it's a red herring, but we've been experiencing a related issue for the past couple months where our coverage on a particular file intermittently fails in two unique ways, in one way it reports one line is missing coverage, and another way it reports 5 or so lines are missing coverage. Locally this is reporting 100% coverage as expected, but the CI build will fail in one of the two ways most of the time. Usually re-running the job once or twice fixes it. That package is using the Very Good Workflows `flutter_package.yml@v1` if that helps narrow our issue at all.

**Comment by danallen88** [comment]
> I'm not sure if it helps or if it's a red herring, but we've been experiencing a related issue for the past couple months where our coverage on a particular file intermittently fails in two unique ways, in one way it reports one line is missing coverage, and another way it reports 5 or so lines are missing coverage. Locally this is reporting 100% coverage as expected, but the CI build will fail in one of the two ways most of the time. Usually re-running the job once or twice fixes it. That package is using the Very Good Workflows `flutter_package.yml@v1` if that helps narrow our issue at all.

For the first time today I was able to reproduce this issue locally after upgrading to `very_good_cli 0.9.1` within the last few days. I am now seeing coverage fail in areas where previously we got 100% coverage locally, and it is failing at the same line that the CI build had been intermittently failing at.

**Comment by alestiago** [comment]
@danallen88 just read this. Is it still an issue?

**Comment by danallen88** [comment]
Hi @alestiago! 👋 I can confirm this is still an issue.

Output:

> Run VeryGoodOpenSource/very_good_coverage@v2
  with:
    path: packages/patient_history_repository/coverage/lcov.info
    min_coverage: 100
  env:
    PUB_CACHE: /home/runner/.pub-cache
    DART_HOME: /opt/hostedtoolcache/dart-sdk
Error: 99.473684[2](https://github.com/IDEXX/vetconnect_plus_mobile/actions/runs/4354861171/jobs/7610700550#step:9:2)10526[3](https://github.com/IDEXX/vetconnect_plus_mobile/actions/runs/4354861171/jobs/7610700550#step:9:3)2 is less than min_coverage [10](https://github.com/IDEXX/vetconnect_plus_mobile/actions/runs/4354861171/jobs/7610700550#step:9:11)0

> Lines not covered:
>   - /home/runner/work/vetconnect_plus_mobile/vetconnect_plus_mobile/packages/patient_history_repository/lib/src/patient_history_repository.dart: 201, 202, 203


Source:
```dart
                      // coverage:ignore-start
                      if (testsWithSameAnalyteCode.length > 1) {
                        // ignore: lines_longer_than_80_chars
                        // TODO(danallen88): how to handle multiple tests here? could have multiple due to multiple runs of the same analyte within this job.
                        _logger.debug(
                            'Found ${testsWithSameAnalyteCode.length} tests for'
                            ' analyte code: ${focusedTest.analyteCode}');
                      }
                      // coverage:ignore-end
```

where the three lines without coverage (201, 202, 203) are:
```dart
_logger.debug(
     'Found ${testsWithSameAnalyteCode.length} tests for'
     ' analyte code: ${focusedTest.analyteCode}');
```

**Comment by alestiago** [comment]
Awesome! Thanks for reaching back @danallen88! 

I'll share this with the team next week. By any chance do you have a minimal reproductive sample?

**Comment by danallen88** [comment]
> By any chance do you have a minimal reproductive sample?

I don't at the moment since I can't reproduce it using the `very_good_cli` locally, only when running it as a GitHub action, so it would require a new project running on a CI pipeline to effectively reproduce. Let me know if it becomes imperative and I can try to find some time to set something up.

**Comment by alestiago** [comment]
@danallen88 I've looked into this, I think this is not an issue with `very_good_coverage` since the action works on an already generated lcov file. So the issue is probably on how the generation of this lcov file.

If you're using Very Good Workflows's Dart package, the lcov generation is done within the workflow itself, see [here](https://github.com/VeryGoodOpenSource/very_good_workflows/blob/405e5d59d41573803e99cb88f9c233a8ab4997c7/.github/workflows/dart_package.yml#L77). There is currently an [open PR](https://github.com/VeryGoodOpenSource/very_good_workflows/pull/107) on Very Good Workflows to support ignoring lines for the dart workflow. Which I think, once merged and released it would solve this issue.

Regarding the issue for Very Good Workflow's Flutter package you mentioned [here](https://github.com/VeryGoodOpenSource/very_good_coverage/issues/200#issuecomment-1360051655) can you open an issue showing reproductive steps in the [Very Good Workflows repository](https://github.com/VeryGoodOpenSource/very_good_workflows/issues)?

Please let me know if my explanation makes sense 😅. If you agree that this is related to Very Good Workflows and not Very Good Coverage we can close this issue. Once again, thanks for opening the issue 💙 !

**Comment by danallen88** [comment]
Thank you for the response @alestiago. I'll keep an eye on the Dart workflow fix and see if that resolves the original issue here. It may take me a bit to come up with a minimal code sample for the Flutter workflow issue but I'll make a note of it and get one out to you all when I can. Thank you for taking a look, and feel free to close this 👍
